widgetry/src/shaders/vertex_140.glsl

```diff
@@ -4,21 +4,39 @@
 uniform vec3 transform;
 // (window width, window height, z value)
 uniform vec3 window;
+// textures grid
+uniform sampler3D textures;
```

Comment:
I'm getting `error: uniform `textures' declared as type `sampler2DArray' and type `sampler3D'`

I changed to `sampler2DArray` and it works fine. Can look into it in the morning, but I'm curious if `sampler2DArray` also works on your machine